File: Maximum Product of Two Elements in an Array.py
# https://leetcode.com/problems/maximum-product-of-two-elements-in-an-array/description/?envType=daily-question&envId=2023-12-12
# Maximum Product of Two Elements in an Array

class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        nums.sort(reverse=True)

        return (nums[0]-1) * (nums[1]-1)

File: Maximum Product of Two Elements in an Array.js
// https://leetcode.com/problems/maximum-product-of-two-elements-in-an-array/?envType=daily-question&envId=2026-07-27
// Maximum Product of Two Elements in an Array

/**
 * @param {number[]} nums
 * @return {number}
 */
var maxProduct = function(nums) {

    let max1 = Math.max(...nums);

    nums.splice(nums.indexOf(max1), 1);

    let max2 = Math.max(...nums);

    return (max1 - 1) * (max2 - 1);

};